web/leaderboard/src/app/page.tsx

```diff
@@ -0,0 +1,446 @@
+"use client";
+
+import { useState, useEffect, useRef, useCallback } from "react";
+import { useQuery } from "@apollo/client";
+import { useAccount } from "wagmi";
+import { useChainContext } from "./queries/ChainContext";
+import Image from "next/image";
+import styles from "./page.module.scss";
+import {
+  Text,
+  GeneralButton,
+  Heading,
+  ArrowTopRight,
+  useClickOutside,
+  toSignificantDecimals,
+} from "@fluidity-money/surfing";
+import Socials from "./components/Socials";
+import Table from "./components/Table";
+import Footer from "./components/Footer";
+import { DropdownOptions } from "./components/Dropdown";
+
+import { Data } from "./types";
+import { tableHeadings, SORTED_ITEM } from "./config";
+
+import {
+  queryLeaderboardRanking24Hours,
+  queryLeaderboardRankingAllTime,
+  queryByUserLeaderboardRanking24Hours,
+} from "./queries/useLeaderboardRanking";
+
+import UseEnsName from "./ensName";
+import { Profile } from "./components/Profile";
+
+export type IRow = React.HTMLAttributes<HTMLDivElement> & {
+  RowElement: React.FC<{ heading: string }>;
+};
+
+export default function Home() {
+  const [filterIndex, setFilterIndex] = useState(0);
+  const [loaded, setLoaded] = useState();
+
+  const { apiState, network, userAddress } = useChainContext();
+  const { address: addressUser, isConnected } = useAccount();
+
+  const {
+    loading: loading24Hours,
+    error: error24Hours,
+    data: data24Hours,
+  } = useQuery(queryLeaderboardRanking24Hours);
+
+  const {
+    loading: loadingAllTime,
+    error: errorAllTime,
+    data: dataAllTime,
+  } = useQuery(queryLeaderboardRankingAllTime);
+
+  const {
+    loading: loadingByUser24Hours,
+    error: errorByUser24Hours,
+    data: dataByUser24Hours,
+  } = useQuery(queryByUserLeaderboardRanking24Hours, {
+    variables: {
+      address: addressUser,
+    },
+  });
+
+  const [sortedData, setSortedData] = useState<Data[]>([]);
+  const [sortedByItem, setSortedByItem] = useState<string>("number");
+
+  const user = {
+    address: String(addressUser),
+    rank: -1,
+    number_of_transactions: 0,
+    volume: 0,
+    yield_earned: "0",
+  } satisfies Data;
+
+  const receiveData24Hours = useCallback(() => {
+    const leaderboard = apiState.leaderboardRanking24Hours.map(
+      (leaderboardRow, i) => ({
+        ...leaderboardRow,
+        rank: i + 1,
+      })
+    );
+    if (isConnected) {
+      if (apiState.dataUser24Hours.length === 0) {
+        apiState.leaderboardRanking24Hours &&
+          apiState.dataUser24Hours &&
+          setSortedData(() => [...leaderboard, user]);
+      } else {
+        const newArray = [...leaderboard, ...apiState.dataUser24Hours];
+        apiState && setSortedData(newArray);
```

Comment:
this could be one line `setSortedData([...leaderboard, ...apiState.dataUser24Hours])`